stackit/internal/services/ske/cluster/resource.go

```diff
@@ -1106,32 +1153,72 @@ func getMaintenanceTimes(ctx context.Context, cl *ske.ClusterResponse, m *Cluste
 	return startTime, endTime, nil
 }
 
-func mapExtensions(cl *ske.ClusterResponse, m *Cluster) {
+func mapExtensions(cl *ske.ClusterResponse, m *Cluster) error {
 	if cl.Extensions == nil || (cl.Extensions.Argus == nil && cl.Extensions.Acl == nil) {
-		return
-	}
-	if m.Extensions == nil {
-		m.Extensions = &Extensions{}
-	}
-	if cl.Extensions.Argus != nil {
-		m.Extensions.Argus = &ArgusExtension{
-			Enabled:         types.BoolPointerValue(cl.Extensions.Argus.Enabled),
-			ArgusInstanceId: types.StringPointerValue(cl.Extensions.Argus.ArgusInstanceId),
-		}
+		m.Extensions = types.ObjectNull(extensionsTypes)
+		return nil
 	}
 
+	var diags diag.Diagnostics
+	acl := types.ObjectNull(aclTypes)
 	if cl.Extensions.Acl != nil {
-		cidr := []attr.Value{}
+		enabled := types.BoolNull()
+		if cl.Extensions.Acl.Enabled != nil {
+			enabled = types.BoolValue(*cl.Extensions.Acl.Enabled)
+		}
+
+		cidrs := []attr.Value{}
 		if cl.Extensions.Acl.AllowedCidrs != nil {
 			for _, v := range *cl.Extensions.Acl.AllowedCidrs {
-				cidr = append(cidr, types.StringValue(v))
+				cidrs = append(cidrs, types.StringValue(v))
 			}
 		}
-		m.Extensions.ACL = &ACL{
-			Enabled:      types.BoolPointerValue(cl.Extensions.Acl.Enabled),
-			AllowedCIDRs: types.ListValueMust(types.StringType, cidr),
+
+		aclValues := map[string]attr.Value{
+			"enabled":       enabled,
+			"allowed_cidrs": types.ListValueMust(types.StringType, cidrs),
```

Comment:
I think we should not use the `...ValueMust` functions outside of tests, as they omit possible errors and panic instead. They also mention it in the description: `"This creation function is only recommended to create List values which will not potentially affect practitioners, such as testing, or exhaustively tested provider logic."`

In this case we can map it to a variable first and return an error in case it fails